### PR TITLE
network policy form editor & script for other resources

### PR DESCRIPTION
**Pull request by andreiv1992**
## Changes

- NetworkPolicy form schema and a script to generate form schemas for other resources

## Fixes

- NetworkPolicy form schema

## How to test it

- Open a network policy file and in the editor you should see a tab where you can edit network policy items in a form

## Screenshots

- https://user-images.githubusercontent.com/5509112/150130162-5543f723-1d4e-4236-b895-dd08073f3dfe.mov

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Comment by andreiv1992** [comment]
If anyone thinks it's worth it I can add documentation for the script and make it more like a CLI, or maybe it should be moved to a separate repo.

File: src/scripts/generate-schema.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
+import fs from 'fs';
+
+const k8Path = `${__dirname}/../../resources/schemas/k8sschemas.json`;
```
Comment (olensmar) [inline src/scripts/generate-schema.ts:3]: use path.sep so this works on windows also!?

**Comment by olensmar** [comment]
> If anyone thinks it's worth it I can add documentation for the script and make it more like a CLI, or maybe it should be moved to a separate repo.

**Comment by olensmar** [comment]
sorry - pressed the wrong button...

File: src/scripts/generate-schema.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    }
+  });
+
+  return {
+    properties: schemaDefinition,
+    propertiesUi: schemaUiDefinition,
+  };
+};
+
+generateSchema('NetworkPolicy');
```
Comment (olensmar) [inline src/scripts/generate-schema.ts:122]: very cool! creating a generic cli sounds like a good idea - it should be able to handle any CRD - not just the bundled k8s ones!?

File: src/scripts/generate-schema.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      }
+
+      const { properties, propertiesUi } = parseFields(kubernetesSchema[childKey].properties);
+      schemaDefinition[key] = {
+        type: 'object',
+        properties,
+      };
+      schemaUiDefinition[key] = {
+        'ui:title': `Object ${key}s`,
+        ...propertiesUi,
```
Comment (olensmar) [inline src/scripts/generate-schema.ts:78]: perhaps you could add `'ui:widget' : 'namespaceSelection' ` to properties names "namespace" ?

File: src/scripts/generate-schema.ts (hunk 2 above)
Comment (andreiv1992) [reply src/scripts/generate-schema.ts]: Yes, as long as it has the same structure

File: src/scripts/generate-schema.ts (hunk 3 above)
Comment (andreiv1992) [reply src/scripts/generate-schema.ts:87]: Added

**Review by olensmar — APPROVED** [review]

**Comment by olensmar** [comment]
Nice one @andreiv1992 !

**Pull request merged**